engine/apps/slack/models/slack_message.py

```diff
@@ -103,12 +117,28 @@ def permalink(self) -> typing.Optional[str]:
 
     @property
     def deep_link(self) -> str:
-        return f"https://slack.com/app_redirect?channel={self.channel_id}&team={self.slack_team_identity.slack_id}&message={self.slack_id}"
+        return f"https://slack.com/app_redirect?channel={self.channel.slack_id}&team={self.slack_team_identity.slack_id}&message={self.slack_id}"
+
+    @classmethod
+    def send_slack_notification(
+        cls, user: "User", alert_group: "AlertGroup", notification_policy: "UserNotificationPolicy"
+    ) -> None:
+        """
+        NOTE: the reason why we pass in `alert_group` as an argument here, as opposed to just doing
+        `self.alert_group`, is that it "looks like" we may have a race condition occuring between two celery tasks:
+        - one which sends out the initial slack message
+        - one which notifies the user (this method) inside of the above slack message's thread
+
+        Still some more investigation needed to confirm this, but for now, we'll pass in the `alert_group` as an argument
+        """
 
-    def send_slack_notification(self, user, alert_group, notification_policy):
```

Comment:
I tried removing `alert_group` here (since `self.alert_group` is a thing..) but it appears that when notifying a user in the Alert Group message thread, the second `SlackMessage` that gets persisted (the one that represents the thread notification message) has a `null` `alert_group` (and I believe it had `null` for another field such as `organization`?)